### PR TITLE
Replace use of click with Argparse and allow yaml specification of the application

### PR DESCRIPTION
**Pull request by danholdaway**
Replace click with Argparse.

Add new API that we discussed allowing loop over applications and single yaml.

Updated the README with some actual details.

### CODE REVIEW

File: src/eva/base.py [hunk 1]
```diff
@@ -109,19 +111,63 @@ def create_and_run(eva_class_name, config, logger=None):
...
+    parser.add_argument('args', nargs='+', type=str, help='Application name [optional] followed ' +
+                        'by the configuration file [madatory]. E.g. eva ObsCorrelationScatter ' +
+                        'conf.yaml')
+
+
+    args = parser.parse_args()
+    args_list = args.args
+
+    # Make sure only 1 or 2 arguments are present
+    assert len(args_list) <= 2, "The maximum number of arguments is two."
```
Comment (CoryMartin-NOAA) [inline src/eva/base.py:151]: What about if `len(0)`?

**Review by CoryMartin-NOAA — COMMENTED** [review]
Just one comment/question, otherwise looks good.

File: src/eva/base.py (hunk 1 above)
Comment (danholdaway) [reply src/eva/base.py:151]: I think `argparse` should take care of that with `nargs='+'`, which ensures at least something is provided in the arguments.

**Review by CoryMartin-NOAA — APPROVED** [review]
Looks great!

**Pull request merged**